### PR TITLE
doc/cephadm: Troubleshooting

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/mgr/cephadm.rst [hunk 1]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
 You can disable this health warning with::
 
   ceph config set mgr mgr/cephadm/warn_on_failed_host_check false
+
+
+Troubleshooting
+---------------
+
+Sometimes, there is a need to investigate why a cephadm command failed or why
```
Comment (asettle) [inline doc/mgr/cephadm.rst]:
```suggestion
Sometimes there is a need to investigate why a cephadm command failed or why

File: doc/mgr/cephadm.rst [hunk 2]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
...
+a specific service no longer runs properly.
+
+As cephadm deploys daemons as containers, troubleshooting daemons is slightly
+different. Here are a few tools and commands to help investigating issues.
+
+Gathering log files
+^^^^^^^^^^^^^^^^^^^
+
+Please use journalctl to gather the log files of all daemons:
```
Comment (asettle) [inline doc/mgr/cephadm.rst]:
```suggestion
Use journalctl to gather the log files of all daemons:

File: doc/mgr/cephadm.rst [hunk 3]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
...
+
+.. note:: Note, By default, cephadm now stores logs in journald. Which means
```
Comment (asettle) [inline doc/mgr/cephadm.rst]:
```suggestion
.. note:: By default cephadm now stores logs in journald. This means

File: doc/mgr/cephadm.rst [hunk 4]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
...
+that you probably won't find damon logs in ``/var/log/ceph/``
```
Comment (asettle) [inline doc/mgr/cephadm.rst]:
```suggestion
that you will no longer find daemon logs in ``/var/log/ceph/``.

File: doc/mgr/cephadm.rst [hunk 5]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
...
+To fetch all state of all daemons of a given host, run::
+
+    fsid="$(cephadm shell ceph fsid)"
+    for name in $(cephadm ls | jq -r '.[].name') ; do
+      systemctl status "ceph-$fsid@$name.service" > $name;
+    done
+
+
+List all downloaded Container images
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (asettle) [inline doc/mgr/cephadm.rst]: I'm not sure what you've got set for RST builds here, but this underline should be the same length as the title

File: doc/mgr/cephadm.rst [hunk 6]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
...
+
+To fetch all state of all daemons of a given host, run::
+
+    fsid="$(cephadm shell ceph fsid)"
+    for name in $(cephadm ls | jq -r '.[].name') ; do
+      systemctl status "ceph-$fsid@$name.service" > $name;
+    done
+
+
+List all downloaded Container images
```
Comment (asettle) [inline doc/mgr/cephadm.rst]:
```suggestion
List all downloaded container images

File: doc/mgr/cephadm.rst [hunk 7]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
...
+
+    podman ps -a --format json | jq '.[].Image'
+    "docker.io/library/centos:8"
+    "registry.opensuse.org/opensuse/leap:15.2"
+
+
+Manually running containers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+cephadm writes small wrappers that run a continers. Please refer to
```
Comment (asettle) [inline doc/mgr/cephadm.rst]:
```suggestion
cephadm writes small wrappers that run a continers. Refer to

File: doc/mgr/cephadm.rst [hunk 8]
```diff
@@ -101,3 +101,68 @@ You can remove a broken host from management with::
...
+``/var/lib/ceph/<cluster-fsid>/<service-name>/unit.run`` to see the command
```
Comment (asettle) [inline doc/mgr/cephadm.rst]:
```suggestion
``/var/lib/ceph/<cluster-fsid>/<service-name>/unit.run`` for the container execution command.

**Comment by sebastian-philipp** [comment]
rebased and fixed.

File: doc/cephadm/administration.rst [hunk 9]
```diff
@@ -180,3 +180,67 @@ Adoption Process
...
+
+    podman ps -a --format json | jq '.[].Image'
+    "docker.io/library/centos:8"
+    "registry.opensuse.org/opensuse/leap:15.2"
+
+
+Manually running containers
+---------------------------
+
+cephadm writes small wrappers that run a continers. Refer to
```
Comment (liewegas) [inline doc/cephadm/administration.rst]: typo: containers

Comment (sebastian-philipp) [reply doc/cephadm/administration.rst]: fixed

**Comment by liewegas** [comment]
retest this please

**Pull request merged**